### PR TITLE
Update dependencies / Clarify parameter override behavior

### PR DESCRIPTION
**Pull request by rossbacher**
Ensure that bundle extra values set in the deeplink method always override values that are set via the URL or via Intent extras.

Added tests to clarify this behavior this.

Update everything except robolectric to latest.

### CODE REVIEW

File: deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/BaseDeepLinkDelegate.kt [hunk 1]
```diff
@@ -304,7 +304,7 @@ open class BaseDeepLinkDelegate @JvmOverloads constructor(
                 if (intent.data == null) {
                     intent.data = sourceIntent.data
                 }
-                intent.putExtras(intentBundle)
+                intent.putExtras(intentBundle.filter { key, _ -> intent.extras?.let { !it.containsKey(key) } ?: true })
```
Comment (elihart) [inline deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/BaseDeepLinkDelegate.kt:307]: I think this is what you want?
```suggestion
                intent.putExtras(intentBundle.filter { key, _ -> intent.extras?.containsKey(key) != true )
```

File: dependencies.gradle [hunk 2]
```diff
@@ -1,20 +1,21 @@
 def versions = [
-    kotlinVersion                : '1.5.31',
-    appCompatVersion             : '1.2.0',
-    localBroadcastManagerVersion : '1.0.0',
+    kotlinVersion                : '1.6.10',
```
Comment (elihart) [inline dependencies.gradle:2]: thanks for all the updates too

File: deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/Utils.kt [hunk 3]
```diff
@@ -23,3 +25,17 @@ fun validateConfigurablePathSegmentReplacements(
             "are:\n${configurablePathSegmentReplacements.keys.joinToString(",\n") { String(it) }}."
     }
 }
+
+/**
+ * Returns a shallow copy of this Bundle that is missing the items for which the predicate returned
+ * false
+ */
+fun Bundle.filter(predicate: (key: String, value: Any?) -> Boolean): Bundle {
```
Comment (elihart) [inline deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/Utils.kt:33]: unit test for this specifically?

**Review by elihart — APPROVED** [review]
thanks!

**Review by mmdango — APPROVED** [review]
Thanks!

File: deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/BaseDeepLinkDelegate.kt (hunk 1 above)
Comment (rossbacher) [reply deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/BaseDeepLinkDelegate.kt:307]: Yeah, I did have it kinda like that yesterday (before reverting to the ugly way of doing it but I could just not wrap my mind around it. Thank you!

**Pull request merged**